src/emqx_router.erl

```diff
@@ -268,9 +268,26 @@ maybe_trans(Fun, Args) ->
 
 -spec(trans(function(), list(any())) -> ok | {error, term()}).
 trans(Fun, Args) ->
-    case mnesia:transaction(Fun, Args) of
-        {atomic, Ok} -> Ok;
-        {aborted, Reason} -> {error, Reason}
+    %% trigger selective receive optimization of compiler,
+    %% ideal for handling bursty traffic.
+    Ref = erlang:make_ref(),
+    Owner = self(),
+    {WPid, RefMon} = spawn_monitor(
+                       fun() ->
+                               Res = case mnesia:transaction(Fun, Args) of
+                                         {atomic, Ok} -> Ok;
+                                         {aborted, Reason} -> {error, Reason}
+                                     end,
+                               Owner ! {Ref, Res}
```

Comment:
we can actually `exit({normal_result, Res}` ?

Reply:
to save one `DOWN` message receive
and no need of two `receive` clauses.

Reply:
I remember there are some side effects of exiting abnormally in VM.  I can save this optimization in later PR if it is not true.